### PR TITLE
Add ORCA documentation

### PR DESCRIPTION
**Pull request by Omar-ORCA**
### Description
Added documentation for ORCA backend
This PR closes NVIDIA/cuda-quantum#1388
It also partially fixes NVIDIA/cuda-quantum#1285

### CODE REVIEW

File: docs/sphinx/using/examples/hardware_providers.rst [hunk 1]
```diff
@@ -47,3 +47,25 @@ The following code illustrates how to run kernels on Quantinuum's backends.
 
    .. literalinclude:: ../../examples/cpp/providers/quantinuum.cpp
       :language: cpp
+
```
[the remark this answers is not in the document]
Comment (Omar-ORCA) [reply docs/sphinx/using/examples/hardware_providers.rst:50]: Added the text "The following code illustrates how to run kernels on ORCA Computing's backends." To the top of ORCA's section
Moved commented out text in the beginning of both the C++ and Python here

File: docs/sphinx/using/backends/hardware.rst [hunk 2]
```diff
@@ -401,6 +401,75 @@ You can then execute a kernel against the platform using the OQC Lucy device
...
+delay lines in series.
+
+.. image:: ../examples/images/orca_tbi.png
+   :width: 400px
+   :align: center
+
+Programmers of CUDA-Q may access the ORCA API from either C++ or Python. There is an environment 
+variable ``ORCA_ACCESS_URL`` that can be set so that the ORCA target can look for it during 
+configuration.
+
```
[the remark this answers is not in the document]
Comment (Omar-ORCA) [reply docs/sphinx/using/backends/hardware.rst:423]: Added  'Setting Credentials' section at the top

File: docs/sphinx/using/backends/hardware.rst [hunk 3]
```diff
@@ -401,6 +401,75 @@ You can then execute a kernel against the platform using the OQC Lucy device
...
+corresponds to 4 photons in 8 modes sent into alternating time-bins in a circuit composed of two 
+delay lines in series.
+
+.. image:: ../examples/images/orca_tbi.png
+   :width: 400px
+   :align: center
+
+Programmers of CUDA-Q may access the ORCA API from either C++ or Python. There is an environment 
+variable ``ORCA_ACCESS_URL`` that can be set so that the ORCA target can look for it during 
+configuration.
```
[the remark this answers is not in the document]
Comment (Omar-ORCA) [reply docs/sphinx/using/backends/hardware.rst:422]: Added example on how to set ORCA_ACCESS_URL environment variable

File: docs/sphinx/examples/cpp/providers/orca.cpp [hunk 4]
```diff
@@ -1,30 +1,95 @@
+// ORCA Computing's PT Series implement the boson sampling model of quantum
+// computation, in which multiple photons are interfered with each other within
+// a network of beam splitters, and photon detectors measure where the photons
+// leave this network.
```
[the remark this answers is not in the document]
Comment (Omar-ORCA) [reply docs/sphinx/examples/cpp/providers/orca.cpp]: Moved commented out text in the beginning of both the C++ and Python to hardware_providers.rst

**Comment by khalatepradnya** [comment]
Suggested patch#1 for spellings:
```
diff --git a/.github/workflows/config/spelling_allowlist.txt b/.github/workflows/config/spelling_allowlist.txt
index d33b9603a..cc4fd98fb 100644
--- a/.github/workflows/config/spelling_allowlist.txt
+++ b/.github/workflows/config/spelling_allowlist.txt
@@ -62,7 +62,6 @@ NVQC
 NVQIR
 OQC
 ORCA
-ORCA's
 OpenACC
 OpenMP
 OpenMPI
@@ -126,7 +125,6 @@ bitstrings
 boolean
 boson
 broadcasted
-bs
 buildable
 callable
 callables
@@ -224,7 +222,6 @@ namespace
 namespaces
 natively
 normalization
-np
 nullary
 optimizer
 optimizers
@@ -241,7 +238,6 @@ prepend
 preprocessor
 probability
 programmatically
-ps
 pybind
 quantize
 quantized
diff --git a/docs/sphinx/examples/cpp/providers/orca.cpp b/docs/sphinx/examples/cpp/providers/orca.cpp
index 4ec859252..bd06a8783 100644
--- a/docs/sphinx/examples/cpp/providers/orca.cpp
+++ b/docs/sphinx/examples/cpp/providers/orca.cpp
@@ -1,7 +1,6 @@
 // Compile and run with:
 // ```
-// nvq++ --target orca --orca-url $ORCA_ACCESS_URL orca.cpp -o out.x
-// && ./out.x
+// nvq++ --target orca --orca-url $ORCA_ACCESS_URL orca.cpp -o out.x && ./out.x
 // ```
 // To use the ORCA Computing target you will need to set the ORCA_ACCESS_URL
 // environment variable or pass the url to the --orca-url flag.
@@ -53,8 +52,10 @@ int main() {
 
   // Optionally, we can also specify the phase shifter angles (created as a
   // linear spaced vector of angles), if the system includes phase shifters
+  // ```
   // std::vector<double> ps_angles(n_beam_splitters);
   // linear_spaced_vector(ps_angles, M_PI / 6, M_PI / 3, n_beam_splitters);
+  // ```
 
   // we can also set number of requested samples
   int n_samples{10000};
@@ -67,8 +68,10 @@ int main() {
   // If the system includes phase shifters, the phase shifter angles can be
   // included in the call
 
+  // ```
   // auto counts = cudaq::orca::sample(input_state, loop_lengths, bs_angles,
   //                                   ps_angles, n_samples);
+  // ```
 
   // Print the results
   counts.dump();
diff --git a/docs/sphinx/examples/python/providers/orca.py b/docs/sphinx/examples/python/providers/orca.py
index a0314c5d6..9e488ea15 100644
--- a/docs/sphinx/examples/python/providers/orca.py
+++ b/docs/sphinx/examples/python/providers/orca.py
@@ -38,7 +38,9 @@ bs_angles = np.linspace(np.pi / 8, np.pi / 3, n_beam_splitters)
 
 # Optionally, we can also specify the phase shifter angles, if the system 
 # includes phase shifters
+# ```
 # ps_angles = np.linspace(np.pi / 6, np.pi / 3, n_beam_splitters)
+# ```
 
 # we can also set number of requested samples
 n_samples = 10000
@@ -50,8 +52,10 @@ counts = cudaq.orca.sample(input_state, loop_lengths, bs_angles, n_samples)
 
 # If the system includes phase shifters, the phase shifter angles can be 
 # included in the call
+# ```
 # counts = cudaq.orca.sample(input_state, loop_lengths, bs_angles, ps_angles, 
 #                            n_samples)
+# ```
 
 # Print the results
 print(counts)

```

**Comment by khalatepradnya** [comment]
Suggested patch#2 for formatting (includes changes from patch#1 above):
Same as applying `yapf`:
```
diff --git a/docs/sphinx/examples/python/providers/orca.py b/docs/sphinx/examples/python/providers/orca.py
index a0314c5d6..be96b3678 100644
--- a/docs/sphinx/examples/python/providers/orca.py
+++ b/docs/sphinx/examples/python/providers/orca.py
@@ -7,7 +7,7 @@ import os
 # To use different targets in the same file, you must update
 # it via another call to `cudaq.set_target()`
 
-# To use the ORCA Computing target you will need to set the ORCA_ACCESS_URL 
+# To use the ORCA Computing target you will need to set the ORCA_ACCESS_URL
 # environment variable or pass a url.
 orca_url = os.getenv("ORCA_ACCESS_URL", "http://localhost/sample")
 
@@ -36,9 +36,11 @@ n_beam_splitters = len(loop_lengths) * len(input_state) - sum(loop_lengths)
 # beam splitter angles
 bs_angles = np.linspace(np.pi / 8, np.pi / 3, n_beam_splitters)
 
-# Optionally, we can also specify the phase shifter angles, if the system 
+# Optionally, we can also specify the phase shifter angles, if the system
 # includes phase shifters
+# ```
 # ps_angles = np.linspace(np.pi / 6, np.pi / 3, n_beam_splitters)
+# ```
 
 # we can also set number of requested samples
 n_samples = 10000
@@ -48,10 +50,12 @@ n_samples = 10000
 # after the job has been returned from ORCA Server.
 counts = cudaq.orca.sample(input_state, loop_lengths, bs_angles, n_samples)
 
-# If the system includes phase shifters, the phase shifter angles can be 
+# If the system includes phase shifters, the phase shifter angles can be
 # included in the call
-# counts = cudaq.orca.sample(input_state, loop_lengths, bs_angles, ps_angles, 
+# ```
+# counts = cudaq.orca.sample(input_state, loop_lengths, bs_angles, ps_angles,
 #                            n_samples)
+# ```
 
 # Print the results
 print(counts)

```

File: docs/sphinx/using/examples/hardware_providers.rst [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Using Quantum Hardware Providers
 -----------------------------------
 
-CUDA-Q contains support for using a set of hardware providers (IonQ, IQM, OQC, and Quantinuum). 
+CUDA-Q contains support for using a set of hardware providers (IonQ, IQM, OQC, Quantinuum and ORCA Computing).
```
Comment (khalatepradnya) [inline docs/sphinx/using/examples/hardware_providers.rst:4]: [consistency / nit]: These are in alphabetical order everywhere else..

Comment (Omar-ORCA) [reply docs/sphinx/using/examples/hardware_providers.rst]: Sorted hardware_providers.rst and hardware.rst

**Comment by schweitzpgi** [comment]
This is blocked waiting for a "code owner".

**Review by khalatepradnya — APPROVED** [review]

**Review by bmhowe23 — APPROVED** [review]
LGTM. Thanks @Omar-ORCA.

**Pull request merged**